node-src/git/getParentCommits.test.ts

```diff
@@ -637,5 +639,123 @@ describe('getParentCommits', () => {
+      expectCommitsToEqualNames(parentCommits, ['D', 'C'], repository);
+    });
+
+    it(`deals with situations where the last build on the squashed branch isn't the last commit`, async () => {
+      // []: has build, <>: is squash merge, (): current commit
+      //
+      //     [B]- D      [branch]
+      //    /
+      //  A -[C]-<E>-(G) [main]
+      const repository = repositories.longLoop;
+      await checkoutCommit('G', 'main', repository);
+      const client = createClient({
+        repository,
+        builds: [
+          ['C', 'main'],
+          ['B', 'branch'],
+        ],
+        // Talking to GH (etc) tells us that commit E is the merge commit for "branch"
+        prs: [['E', 'branch']],
+      });
+      const git = { branch: 'main', ...(await getCommit()) };
+
+      const parentCommits = await getParentCommits({ client, log, git, options } as any);
+      expectCommitsToEqualNames(parentCommits, ['C', 'B'], repository);
+    });
+
+    it(`deals with situations where squashed branches have no builds`, async () => {
+      // []: has build, <>: is squash merge, (): current commit
+      //
+      //     B - D       [branch]
+      //    /
+      //  A -[C]-<E>-(G) [main]
+      const repository = repositories.longLoop;
+      await checkoutCommit('G', 'main', repository);
+      const client = createClient({
+        repository,
+        builds: [
+          ['C', 'main'],
+        ],
+        // Talking to GH (etc) tells us that commit E is the merge commit for "branch"
+        prs: [['E', 'branch']],
+      });
+      const git = { branch: 'main', ...(await getCommit()) };
+
+      const parentCommits = await getParentCommits({ client, log, git, options } as any);
+      expectCommitsToEqualNames(parentCommits, ['C'], repository);
+    });
+
+    it(`deals with situations where squashed branches no longer exist in the repo but have a build`, async () => {
+      // []: has build, <>: is squash merge, (): current commit
+      //
+      //  [B] [no longer in repo]
+      //
+      //  [A]- C -<(D)> [main]
+      const repository = repositories.twoRoots;
+      await checkoutCommit('D', 'main', repository);
+      const client = createClient({
+        repository,
+        builds: [
+          ['A', 'main'],
+          ['B', 'xxx'],
+        ],
+        // Talking to GH (etc) tells us that commit D is the merge commit for "branch" (which no longer exists)
+        prs: [['D', 'branch']],
+      });
+      const git = { branch: 'main', ...(await getCommit()) };
+
+      const parentCommits = await getParentCommits({ client, log, git, options } as any);
+      // This is A and not B because we do not know anything about the deleted branch and its commits
+      expectCommitsToEqualNames(parentCommits, ['A'], repository);
+    });
+
+    it(`deals with situations where squashed branches themselves have squash merge commits`, async () => {
```

Comment:
I think this title is misleading. We explicitly *don't* deal with this (in the sense that we don't recurse and find `F` as a baseline).